File: java/altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/graph/v2/JRank.java
package org.cjoakim.cosmos.altgraph.data.common.graph.v2;

import lombok.Data;

/**
 * Instances of this class are sortable results of JGraphT PageRank calls.
 * <p>
 * Chris Joakim, Microsoft, November 2022
 */

@Data
public class JRank {

    private String key;
    private Double value;

    public JRank(String key, Double value) {
        super();
        this.key = key;
        this.value = value;
    }

    public int compareTo(Object other) {

        JRank otherInstance = (JRank) other;

        if (this.getValue() > otherInstance.getValue()) {
            return -1;
        } else if (this.getValue() < otherInstance.getValue()) {
            return 1;
        } else {
            return 0;
        }
    }
}

File: dotnet/src/altgraph_shared_app/Services/Graph/v2/JRank.cs
namespace altgraph_shared_app.Services.Graph.v2
{
  public class JRank
  {
    public string Key { get; set; } = string.Empty;
    public double Value { get; set; }

    public JRank(string key, double value)
    {
      Key = key;
      Value = value;
    }

    public int CompareTo(Object other)
    {
      JRank otherInstance = (JRank)other;

      if (Value > otherInstance.Value)
      {
        return -1;
      }
      else if (Value < otherInstance.Value)
      {
        return 1;
      }
      else
      {
        return 0;
      }
    }
  }
}